test/dynamo/test_misc.py

```diff
@@ -3549,6 +3549,20 @@ def fn(a, b):
         self.assertEqual(cnts.frame_count, 1)
         self.assertEqual(cnts.op_count, 2)
 
+    def test_set_grad_from_input_to_intermed(self):
+        param = torch.rand(2, 3)
+
+        def f(p):
+            p.grad = torch.rand_like(p)
+            p.grad = p.grad.to_sparse()
+            return p.grad
+
+        cnts = torch._dynamo.testing.CompileCounter()
+        compiled_f = torch._dynamo.optimize(cnts)(f)
+        compiled_f(param)
```

Comment:
it would be nice to test that this actually has the correct result, if I undo your PR and modify the counts this test "passes"